exp/api/v1beta1/machinepool_types.go

```diff
@@ -38,6 +38,7 @@ type MachinePoolSpec struct {
 
 	// Number of desired machines. Defaults to 1.
 	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
```

Comment:
I know we discussed this in the past. But since we are here do we need to "distinguish between explicit zero and not specified."? That's never the case since we are defaulting 1, at least for MachineSets and MachineDeployments. I can't see default marker here.

Reply:
> Or even better:
	// Number of desired machines. Defaults to 1.
	// +kubebuilder:default=1
         Replicas int32 `json:"replicas,omitempty"`

Adding `// +kubebuilder:default=1` could be a good idea. But making it non-pointer probably involves more code changes (idk breaking or not). With this PR, trying to make as minimal changes and best effort compliance with the API conventions.

@sbueringer can add more details

Reply:
@enxebre Would it be okay for you to move this conversation over to #5304? We try to keep this PR as non-breaking as possible (i.e. we only add +optional to fields which already have omitempty, because this doesn't have any consequences)

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/5304#issuecomment-925846921

Reply:
Thx!